lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp

```diff
@@ -1412,16 +1411,24 @@ void mlir::torch::onnx_c::populateDefaultDomainAtoF(
             binder.tensorResultType(resultType))
           return failure();
 
+        auto operandTy = cast<Torch::ValueTensorType>(operand.getType());
+        llvm::SmallVector<int64_t> shape(operandTy.getSizes());
+        int64_t rank = shape.size();
+
         // If axis is negative, count from the right instead of left
-        int64_t rank =
-            cast<Torch::ValueTensorType>(operand.getType()).getSizes().size();
         if (axis < 0)
           axis = rank + axis;
 
-        Value collapsedRight;
-        auto baseType = Torch::ValueTensorType::getWithLeastStaticInformation(
```

Comment:
I didn't notice this in my review of this the first time around, should have. One question about the replacement code is, does it work with dynamic dimensions? Should it be something like 

 for (int i = axis + 1; i < rank; ++i) {
          if (shape[i] == kUnknownSize) { shape[axis] = kUnknownSize; break;  }
          shape[axis] *= shape[i];
}

Reply:
If `axis == rank` then the loop never executes meaning we never index into `shape`.